apps/src/studio/studio.js

```diff
@@ -3153,6 +3164,19 @@ Studio.setItemActivity = function (opts) {
     Studio.items.forEach(function (item) {
       if (item.className === opts.className) {
         item.setActivity(opts.type);
+
+        // For verifying success, record this combination of activity type and
+        // item type.
+
+        if (!Studio.trackedBehavior.setActivityRecord) {
+          Studio.trackedBehavior.setActivityRecord = [];
+        }
+
+        if (!Studio.trackedBehavior.setActivityRecord[opts.className]) {
+          Studio.trackedBehavior.setActivityRecord[opts.className] = [];
+        }
+
+        Studio.trackedBehavior.setActivityRecord[opts.className][opts.type] = true;
```

Comment:
(1) Seems odd to do this inside the forEach() loop - you'll end up setting the record a bunch of times if there are many items of that class - and not at all if there aren't any items present.
(2) I wonder if the goal here is to judge the "end state" (are there items present? what is the activity state of this item class?) or the "API history" (did this particular API get called with these specific parameters?) - this implementation is kind of a combination of the two.